lib/src/store.dart

```diff
@@ -58,22 +60,60 @@ class Store {
   /// As an example, [transformer] could be used to throttle the number of
   /// triggers this [Store] emits for state that may update extremely frequently
   /// (like scroll position).
-  Store.withTransformer(StreamTransformer<dynamic, dynamic> transformer) {
-    _streamController = new StreamController<Store>();
+  Store.withTransformer(StreamTransformer<dynamic, dynamic> transformer)
+      : _streamController = new StreamController<Store>() {
+    manageStreamController(_streamController);
 
     // apply a transform to the stream if supplied
     _stream = _streamController.stream
         .transform(transformer as StreamTransformer<Store, dynamic>)
         .asBroadcastStream() as Stream<Store>;
   }
 
+  /// Adds a subscription to this `Store`.
+  ///
+  /// Each time this `Store` triggers (by calling [trigger]), indicating that
+  /// data has been mutated, [onData] will be called.
+  ///
+  /// If the `Store` has been disposed, this method throws a [StateError].
+  ///
+  /// It is the caller's responsibility to cancel the subscription when
+  /// needed.
+  StreamSubscription<Store> listen(StoreHandler onData,
+      {Function onError, void onDone(), bool cancelOnError}) {
+    if (isDisposed) {
+      throw new StateError('Store has been disposed');
+    }
+
+    return _stream.listen(onData,
+        onError: onError, onDone: onDone, cancelOnError: cancelOnError);
+  }
+
+  /// Registers an [ActionSubscription] to be canceled when the store is disposed.
+  ///
+  /// This supports the following pattern for consumers:
+  ///
+  ///     manageActionSubscription(myAction.listen(_myHandler));
+  ///
+  @mustCallSuper
+  @protected
+  void manageActionSubscription(ActionSubscription subscription) {
+    manageDisposer(() => new Future(() {
```

Comment:
The reason we have to provide a `Future` is basically an artifact of the fact that the `cancel` method on `StreamSubscription` returns a `Future` (or `null`) but is **not** async. So to make everything make sense, I just require anything that wants to dispose to return a future. That way you **can** do async disposal, but you don't have to, and we can handle cases like `StreamSubscription` as well.